src/components/templates/discount-table.tsx

```diff
@@ -0,0 +1,368 @@
+  //       : parseInt(offset) - parseInt(limit)
+  //   const baseUrl = qs.parseUrl(window.location.href).url
+
+  //   const queryParts = {
+  //     q: query,
+  //     offset: updatedOffset,
+  //     limit,
+  //     ...filtersOnLoad,
+  //   }
+
+  //   const prepared = qs.stringify(queryParts, {
+  //     skipNull: true,
+  //     skipEmptyString: true,
+  //   })
+
+  //   console.log("baseurl", baseUrl)
+  //   window.history.replaceState(baseUrl, "", `?${prepared}`)
+
+  //   refresh({
+  //     search: {
+  //       ...queryParts,
+  //       is_dynamic: showDynamic,
+  //     },
+  //   }).then(() => {
+  //     setOffset(updatedOffset)
+  //   })
+  // }
+
+  const duplicateDiscount = (discount) => {
+    setLoading(true)
+    const newRule = {
+      description: discount.rule.description,
+      type: discount.rule.type,
+      value: discount.rule.value,
+      allocation: discount.rule.allocation,
+      valid_for: discount.rule.valid_for.map((product) => product.id),
+    }
+    const newDiscount = {
+      code: `${discount.code} DUPLICATE`,
+      is_dynamic: discount.isDynamic,
+      rule: newRule,
+      starts_at: discount.starts_at,
+      ends_at: discount.ends_at,
+      regions: discount.regions.map((region) => region.id),
+      valid_duration: discount.valid_duration,
+      usage_limit: discount.usage_limit,
+      is_disabled: discount.is_disabled,
+      metadata: discount.metadata,
+    }
+
+    Medusa.discounts
+      .create(newDiscount)
+      .then(() => {
+        toaster("Successfully created discount", "success")
+        handleCheckbox()
+      })
+      .catch((error) => {
+        toaster(getErrorMessage(error), "error")
+      })
+      .finally(() => setLoading(false))
+  }
+
+  const filteringOptions = [
+    {
+      title: "Type",
+      options: [
+        {
+          title: "All",
+          count: count,
+          onClick: () => searchQuery(),
+        },
+        {
+          title: "Percentage",
+          onClick: () => searchQuery({ rule: { type: "percentage" } }),
+        },
+        {
+          title: "Fixed Amount",
+          onClick: () =>
+            searchQuery({ rule: { type: "fixed" }, is_dynamic: false }),
+        },
+        {
+          title: "Dynamic",
+          onClick: () => {
+            searchQuery({ is_dynamic: true })
+          },
+        },
+      ],
+    },
+    {
+      title: "Status",
+      options: [
+        {
+          title: "All",
+          count: count,
+          onClick: () => searchQuery(),
+        },
+        {
+          title: "Active",
+          count: 0,
+          onClick: () => {},
+        },
+        {
+          title: "Draft",
+          count: 0,
+          onClick: () => {},
+        },
+        {
+          title: "Expired",
+          count: 0,
+          onClick: () => {},
+        },
+        {
+          title: "Scheduled",
+          count: 0,
+          onClick: () => {},
+        },
+      ],
+    },
+  ]
```

Comment:
Should we delete these for now? I could imagine that these would be refactored when we start migrating to `react-table`